### ISSUE TITLE
Regional instances force use of point-in-time recovery

### ISSUE THREAD

**Issue by jawnsy** [terraform-google-modules/terraform-google-sql-db#344]
### TL;DR

The module has logic that force-enables the Point-in-Time Recovery feature on, but this does not seem to be a requirement of GCP itself

### Expected behavior

I expected the module to be consistent with Google's implementation. Given that GCP allows HA instances without PITR, the module should allow the same.

### Terraform Configuration

```hcl
n/a
```

### Terraform Version

```sh
n/a
```

### Additional information

The code enforcing this logic is here: https://github.com/terraform-google-modules/terraform-google-sql-db/blob/cf8fc4c637ecd75b16b8ff6fe903e2da587639f0/modules/postgresql/main.tf#L34-L36

I was able to create an instance in HA module with PITR disabled:

![image](https://user-images.githubusercontent.com/52710/191076260-a0598051-7599-4868-8192-21120b2095ce.png)

It's possible that the Console UI is at fault here, since the provider documentation says:

>[availability_type](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/sql_database_instance#availability_type) - (Optional, Default: ZONAL) The availability type of the Cloud SQL instance, high availability (REGIONAL) or single zone (ZONAL).' For all instances, ensure that settings.backup_configuration.enabled is set to true. For MySQL instances, ensure that settings.backup_configuration.binary_log_enabled is set to true. For Postgres instances, ensure that settings.backup_configuration.point_in_time_recovery_enabled is set to true.

This may be interpreted as "regional databases require PITR to be enabled", in which case the module behavior is correct
